### PR TITLE
Contributing Guidelines

### PR DESCRIPTION
**Pull request by jmshrv**
This PR adds some basic contributing guidelines, mostly related to setting up a development environment. It covers most of the weird bits of working on Finamp, I can't think of anything more to say.

(pubspec.lock was also updated lol)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+* Hive - the database that Finamp uses for storing all data
+* `json_serializable` - For deserialising JSON into classes
+* Chopper - For talking to Jellyfin over HTTP
+    * This layer (`lib/services/jellyfin_api.dart`) is not used by the app directly. The user-facing API is located at `lib/services/jellyfin_api_helper.dart`.
+
+To rebuild these files, run `dart run build_runner build --delete-conflicting-outputs`. This must be done when:
+
+* Modifying a class that is returned by Jellyfin (such as the classes in `lib/models/jellyfin_models.dart`)
+* Adding fields to a database class (annotated with `@HiveType`)
```
Comment (Chaphasilor) [inline CONTRIBUTING.md:27]: Maybe add a section that describes common issues that could be caused by not rebuilding this? E.g. settings not saving, hive errors on startup, missing types, missing data for fields in JSON responses, etc.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+### Hive
+
+As said earlier, Finamp uses Hive for all data storage needs. If you're doing work that involves data storage, I recommend you read [the Hive docs](https://docs.hivedb.dev/#/). Please ensure that your changes work when upgrading Finamp from the current release to your changes, as not handling upgrades will cause the app to crash.
+
```
Comment (Chaphasilor) [inline CONTRIBUTING.md:32]: Explain that (on Android at least) you can and need to clear the app storage (or uninstall the app) when you want to downgrade / switch to a version that is missing some hive types

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## The Redesign
+
+The biggest main piece of work being done on Finamp at the moment is the redesign. The relevant issue can be found [here](https://github.com/jmshrv/finamp/issues/220). The `redesign` branch has diverged a lot from `main`, but I try to keep it updated. If you're struggling to decide what to work on, the redesign is a good place to look :)
+
```
Comment (Chaphasilor) [inline CONTRIBUTING.md:36]: We need to add the link to the discord server here (once the beta is live)

File: pubspec.lock [hunk 4]
```diff
@@ -1173,5 +1165,5 @@ packages:
     source: hosted
     version: "3.1.2"
 sdks:
-  dart: ">=3.1.0 <4.0.0"
-  flutter: ">=3.13.0"
+  dart: ">=3.0.0 <4.0.0"
+  flutter: ">=3.7.0"
```
Comment (Chaphasilor) [inline pubspec.lock:1169]: Are you sure this won't cause any issues?

**Review by Chaphasilor — CHANGES_REQUESTED** [review]
Also mention the need to register TypeAdapters (and how to do that) when adding new hive types

File: CONTRIBUTING.md (hunk 1 above)
Comment (jmshrv) [reply CONTRIBUTING.md:27]: Added in 82bf037

File: CONTRIBUTING.md (hunk 3 above)
Comment (jmshrv) [reply CONTRIBUTING.md:42]: Will do, when it's ready :)

**Comment by jmshrv** [comment]
> Also mention the need to register TypeAdapters (and how to do that) when adding new hive types

Fixed in 0aef744

File: README.md [hunk 5]
```diff
@@ -20,15 +20,9 @@ Note: The F-Droid release may take a day or two to get updates because since [F-
...
+## Contributing
 
-<a href="https://hosted.weblate.org/engage/finamp/">
-<img src="https://hosted.weblate.org/widgets/finamp/-/finamp/multi-auto.svg" alt="Translation status" />
-</a>
-
-Finamp uses Weblate to manage translations: **https://hosted.weblate.org/engage/finamp/**
-
-Feel free to add new languages if yours isn't there yet. If you have any questions, such as the context of a string, you can ask in the [Translation Discussions](https://github.com/jmshrv/finamp/discussions/categories/translations).
+Finamp's contributing guidelines can be found in [CONTRIBUTING.md](CONTRIBUTING.md)
```
Comment (Chaphasilor) [inline README.md:25]: Maybe something a bit more motivational would be nice.  
> Finamp is a community-driven project and relies on people like **you** and their contributions. To learn how you could help out with making Finamp even better, take a look at our [Contribution Guidelines](CONTRIBUTING.md)

**Review by Chaphasilor — APPROVED** [review]
Looks good overall!

**Comment by Chaphasilor** [comment]
I went ahead and merged it. It's definitely better than nothing, and right now everything looks good to me 😁

**Pull request merged**